clients/rospy/src/rospy/core.py

```diff
@@ -149,6 +151,57 @@ def rospywarn(msg, *args):
 logerror = logerr # alias logerr
 
 logfatal = logging.getLogger('rosout').critical
+
+
+class LoggingThrottle(object):
+
+    last_logging_time_table = {}
+
+    def __call__(self, id, logging_func, period, msg):
+        """Do logging specified message periodically.
+
+        - id (str): Id to identify the caller
+        - logging_func (function): Function to do logging.
+        - period (float): Period to do logging in second unit.
+        - msg (object): Message to do logging.
+        """
+        now = rospy.Time.now()
+
+        last_logging_time = self.last_logging_time_table.get(id)
```

Comment:
Maybe I need to get more detailed position where the function is called.
Currently, the `inspect.stack()[1][1:]` returns only line where the function is called:

```
% cat spam.py
import inspect


def loginfo_throttle():
    print(inspect.stack()[1])


def main():
    loginfo_throttle(); loginfo_throttle()


if __name__ == '__main__':
    main()

% python spam.py 
(<frame object at 0x7f3d6b2ed3a0>, 'spam.py', 9, 'main', ['    loginfo_throttle(); loginfo_throttle()\n'], 0)
(<frame object at 0x7f3d6b2ed3a0>, 'spam.py', 9, 'main', ['    loginfo_throttle(); loginfo_throttle()\n'], 0)
```